### ISSUE TITLE
Using the function's code which is memoized as part of the cache key hashing

### ISSUE THREAD

**Issue by komailrk** [sh4nks/flask-caching#154]
When using memoize the function in question could have the source changed but the cache key generated will remain the same. The idea is to include the function's source code in cache key hash to avoid returning the data when the deriving mechanism has been modified.

To avoid any issues, this option should be introduced as an optional feature which can be enabled on demand when needed.

For example:
```
@app_cache.memoize(timeout=600, source_check=True)
def my_function(arg1):
...
```

To reproduce the issue:

1. Create a function with memoize decorator
2. Run the function either by calling the URL or however way possible
3. Change the source code such that the return value will be different provided the same input parameters
4. Run the function again and the return value will be from the stale function

I can start a PR and introduce this feature. Would like to get some feedback on this idea.

Using Python `inspect` the source code can be retrieved and added to the cache key.

**Comment by gergelypolonkai** [comment]
I suspect either an edge case or an XY problem here. Could you elaborate why you need this functionality?

The main problem is, function code can be really long, possibly not fitting in a cache key. What do you have in mind to mitigate this?

**Comment by komailrk** [comment]
Hi @gergelypolonkai,

I have a working solution which uses the function's code which includes any decorators and arguments the function accepts, hashes the code and includes it in the cache key.

```
           cache_key = hash_method()
            cache_key.update(updated.encode("utf-8"))

            if callable(f) and source_check:
                source_code = inspect.getsource(f)
                cache_key.update(source_code.encode("utf-8"))

            cache_key = base64.b64encode(cache_key.digest())[:16]
            cache_key = cache_key.decode("utf-8")
```

Here is the part of the code explaining my proposal.

**Comment by gergelypolonkai** [comment]
OK, it seems reasonable then, can you turn into a PR?

In my head, though, this should be globally configurable, like:

```
def cached(…, source_check=None):
    …
    if source_check is None:
        source_check = config['CACHE_SOURCE_CHECK']

    if source_check:
        # Do your magic
```

So if you want to source check all functions, just turn it on in the config. If you want to exclude one function for a reason, just add `source_check=False` to the arguments.

Also, you should add the same functionality to the `cache` method, not just `memoize`.

**Comment by sh4nks** [comment]
Merged your PR #156.

Thanks for the PR!
